syntax = "proto2";

package drones.lp_solver;

message VariableDesc {
  enum VariableType {
    TOTAL_SCORE = 0;         // Double
    ORDER_SCORE = 1;         // Double
    ORDER_COMPLETENESS = 2;  // Double
    ORDER_STATE = 3;         // Int
    DRONE_STATE = 4;         // Int
    DRONE_LOC = 5;           // Bool
    WAREHOUSE_STATE = 6;     // Int

    WAIT = 7;      // Bool
    LOAD = 8;      // Bool
    UNLOAD = 9;    // Bool
    DELIVER = 10;  // Bool
  }
  optional VariableType type = 1;
  optional int32 drone = 2;
  // Locations are enumerated in the following way: warehouse locations are
  // enumerated by warehouse IDs and then order locations are appended.
  // Example: Let's say there are 5 warehouses and 10 orders, so 15 locations,
  //          enumerated as [0, 14] where are [0, 4] are locations of the
  //          warehouses [0, 4], respectively and [5, 14] are locations of the
  //          orders [0, 9], respectively.
  optional int32 location = 3;
  optional int32 t = 4;
  optional int32 warehouse = 5;
  optional int32 product = 6;
  optional int32 order = 7;
  optional int32 num_items = 8;
}
